parser-typechecker/src/Unison/Util/Pretty.hs

```diff
@@ -668,6 +670,20 @@ string = fromString
 shown :: (Show a, IsString s) => a -> Pretty s
 shown = fromString . show
 
+softHang :: (LL.ListLike s Char, IsString s)
+  => Pretty s
+  -> Pretty s
+  -> Pretty s
+softHang from = softHang' from "  "
+
+softHang' :: (LL.ListLike s Char, IsString s)
+  => Pretty s
+  -> Pretty s
+  -> Pretty s
+  -> Pretty s
+softHang' from by p = group $
+  (from <> " " <> group p) `orElse` (from <> "\n" <> group (indent by p))
```

Comment:
Maybe a comment on what these do and how they differ from `hang` would be good, along with an example or two.